src/dispatch/auth/permissions.py

```diff
@@ -332,6 +332,26 @@ def has_required_permissions(
                 return True
 
 
+class IncidentCommanderOrReporterPermission(BasePermission):
+    def has_required_permissions(
+        self,
+        request: Request,
+    ) -> bool:
+        current_user = get_current_user(request=request)
+        pk = PrimaryKeyModel(id=request.path_params["incident_id"])
+        current_incident = incident_service.get(db_session=request.state.db, incident_id=pk.id)
+        if not current_incident:
+            return False
+
+        if current_incident.commander:
+            if current_incident.commander.individual.email == current_user.email:
+                return True
+
+        if current_incident.reporter:
+            if current_incident.reporter.individual.email == current_user.email:
+                return True
+
```

Comment:
```suggestion
if current_incident.commander and current_incident.commander.individual.email == current_user.email:
    return True

if current_incident.reporter and current_incident.reporter.individual.email == current_user.email:
    return True

return False
```

If none of the conditions meet the function will return `None` which I think is not expected.